src/containers/profile/info.svelte

```diff
@@ -1,70 +1,88 @@
 <script>
-  import { createEventDispatcher } from 'svelte';
-  import Button from 'ui/button.svelte';
-  import TextField from 'ui/text-field.svelte';
-  import Dropdown from 'ui/dropdown.svelte';
+  import { stores } from '@sapper/app';
+  import { Button, TextField, DropdownShell, Dropdown, H1 } from 'attractions';
+  import ReclaimInnopointsModal from '@/components/profile/reclaim-innopoints-modal.svelte';
+  import * as api from '@/utils/api.js';
 
   export let account;
+  const telegramUsernameRegex = /[A-Za-z0-9_]{5,32}/;
+
+  const { session } = stores();
 
   let tgChange = false;
   let tgUsername = account.telegram_username || '';
+  $: usernameValid = telegramUsernameRegex.test(tgUsername);
+  let reclaimInnopointsModalOpen = false;
 
-  function saveUsername() {
-    if (tgUsername !== '' && tgUsername !== account.telegram_username) {
-      dispatch('username-change', tgUsername);
+  async function saveUsername() {
+    if (tgUsername === '' || tgUsername === account.telegram_username) {
```

Comment:
```suggestion
    if (!usernameValid || tgUsername === account.telegram_username) {
```

Reply:
That thing actually serves a different purpose. If the username isn't valid, this function won't be even called. Amended